### DISCUSSION TITLE
Adding support for image vision

### DISCUSSION THREAD

**Discussion by rPman** [ggml-org/llama.cpp#1115] [Ideas]
This https://github.com/Vision-CAIR/MiniGPT-4 project used vikuna (llama based weights) model for training (frosen weights, trained only first layer, as i understand).

**Comment by luussta** [comment]
It's really a good idea.

**Comment by Don-Chad** [comment]
I agree!

Llava is even better - better training data and performance. Plus also already has many good quantised models available.

https://huggingface.co/Hyunel/llava-13b-v1-1-4bit-128g/

Would anyone be able to describe what would be needed tot add image support to llama.cpp? Myself U can only do python unfortunately, but am really curious how hard it would be.

**Comment by jonahkaye** [comment]
Would acceleration be possible also for tesseract https://github.com/tesseract-ocr/tesseract ?

**Comment by Green-Sky** [comment]
duplicate (mostly) of https://github.com/ggerganov/llama.cpp/discussions/1050
